persistence/src/test/scala/com/evolutiongaming/akkaeffect/persistence/EventSourcedActorOfTest.scala

```diff
@@ -0,0 +1,1100 @@
+package com.evolutiongaming.akkaeffect.persistence
+
+import java.time.Instant
+import akka.actor.{ActorIdentity, ActorRef, ActorSystem, Identify}
+import akka.persistence.Recovery
+import akka.testkit.TestActors
+import cats.data.{NonEmptyList => Nel}
+import cats.effect.kernel.Ref
+import cats.effect.{Async, Deferred, IO, Resource, Sync, Temporal}
+import cats.effect.syntax.all._
+import cats.effect.unsafe.implicits.global
+import cats.syntax.all._
+import com.evolutiongaming.akkaeffect.IOSuite._
+import com.evolutiongaming.akkaeffect.persistence.InstrumentEventSourced.Action
+import com.evolutiongaming.akkaeffect.testkit.Probe
+import com.evolutiongaming.akkaeffect.{ActorSuite, _}
+import com.evolutiongaming.catshelper.CatsHelper._
+import com.evolutiongaming.catshelper.{FromFuture, ToFuture, ToTry}
+import org.scalatest.funsuite.AsyncFunSuite
+import org.scalatest.matchers.should.Matchers
+
+import scala.concurrent.duration._
+import scala.reflect.ClassTag
+
+class EventSourcedActorOfTest extends AsyncFunSuite with ActorSuite with Matchers {
+
+  test("all") {
+    `persistentActorOf`[IO](actorSystem).run()
+  }
+
+  test("recover with empty state") {
+    `recover with empty state`[IO](actorSystem).run()
+  }
+
+  test("recover from snapshot") {
+    `recover from snapshot`[IO](actorSystem).run()
+  }
+
+  test("recover from snapshot with deleted events") {
+    `recover from snapshot with deleted events`[IO](actorSystem).run()
+  }
+
+  test("recover from events") {
+    `recover from events`[IO](actorSystem).run()
+  }
+
+  test("recover from events with deleted snapshot") {
+    `recover from events with deleted snapshot`[IO](actorSystem).run()
+  }
+
+  test("recover from snapshot and events") {
```

Comment:
The test is flaky in CI. I did not check why, but it uses some `IO.sleep` statements, so it might be worth to consider alternative.

https://github.com/evolution-gaming/akka-effect/actions/runs/7505262506/job/20434145553?pr=276#step:5:220

Reply:
I will try to fix the test later, its too flaky

Reply:
I approved PR as all other issues were resolved, but you might want to fix this test first, before merging, as it will be blocking pipeline if it breaks all the time.

Reply:
The issue was in implementation and is fixed by https://github.com/evolution-gaming/akka-effect/pull/276/commits/755241ae9147dae22dc40302341eef2aea877da7